test/boost/tablets_test.cc

```diff
@@ -1396,3 +1396,26 @@ SEASTAR_THREAD_TEST_CASE(test_load_balancing_with_random_load) {
     }
   }).get();
 }
+
+SEASTAR_TEST_CASE(test_tablet_id_and_range_side) {
+    static constexpr size_t tablet_count = 128;
+    locator::tablet_map tmap(tablet_count);
+
+    for (size_t id = 0; id < tablet_count; id++) {
+        auto tr = tmap.get_token_range(tablet_id(id));
+
+        {
+            auto token = tr.start()->value() == dht::minimum_token() ? dht::first_token() : tr.start()->value();
+            auto [tid, side] = tmap.get_tablet_id_and_range_side(next_token(token));
+            BOOST_REQUIRE_EQUAL(tid.value(), id);
+            BOOST_REQUIRE_EQUAL(side, tablet_range_side::left);
+        }
+        {
+            auto [tid, side] = tmap.get_tablet_id_and_range_side(tr.end()->value());
+            BOOST_REQUIRE_EQUAL(tid.value(), id);
+            BOOST_REQUIRE_EQUAL(side, tablet_range_side::right);
+        }
```

Comment:
Also good to test: 
 - expected end of the left group
 - expected start of the right group